apps/api/src/app/change/e2e/promote-layout-changes.e2e.ts

```diff
@@ -0,0 +1,343 @@
+import {
+  ChangeEntityTypeEnum,
+  ITemplateVariable,
+  LayoutDescription,
+  LayoutId,
+  LayoutName,
+  TemplateVariableTypeEnum,
+} from '@novu/shared';
+import { UserSession } from '@novu/testing';
+
+describe('Promote Layout Changes', () => {
+  let session: UserSession;
+  const changeRepository: ChangeRepository = new ChangeRepository();
+  const layoutRepository = new LayoutRepository();
+  const environmentRepository: EnvironmentRepository = new EnvironmentRepository();
+
+  beforeEach(async () => {
+    session = new UserSession();
+    await session.initialize();
+  });
+
+  it('should promote a new layout created to production', async () => {
+    const layoutName = 'layout-name-creation';
+    const layoutDescription = 'Amazing new layout';
+    const content = '<html><body><div>Hello {{organizationName}} {{{body}}}</div></body></html>';
+    const variables = [
+      { name: 'organizationName', type: TemplateVariableTypeEnum.STRING, defaultValue: 'Company', required: false },
+    ];
+    const isDefault = true;
+
+    const createLayoutPayload = {
+      name: layoutName,
+      description: layoutDescription,
+      content,
+      variables,
+      isDefault,
+    };
+
+    const {
+      body: {
+        data: { _id: layoutId },
+      },
+    } = await session.testAgent.post('/v1/layouts').send(createLayoutPayload);
+
+    expect(layoutId).to.be.ok;
+
+    const {
+      body: { data: devLayout },
+    } = await session.testAgent.get(`/v1/layouts/${layoutId}`);
+
+    const changes = await changeRepository.find(
+      {
+        _environmentId: session.environment._id,
+        _organizationId: session.organization._id,
+        enabled: false,
+        _entityId: layoutId,
+        type: ChangeEntityTypeEnum.DEFAULT_LAYOUT,
+      },
+      '',
+      {
+        sort: { createdAt: 1 },
+      }
+    );
+
+    expect(changes.length).to.eql(1);
+    expect(changes[0]._entityId).to.eql(layoutId);
+    expect(changes[0].type).to.eql(ChangeEntityTypeEnum.DEFAULT_LAYOUT);
+    expect(changes[0].change).to.deep.include({ op: 'add', path: ['_id'], val: layoutId });
+
+    await session.applyChanges({
+      enabled: false,
+    });
+
+    const prodEnv = await getProductionEnvironment();
+
+    const prodLayout = await layoutRepository.findOne({
+      _environmentId: prodEnv._id,
+      _parentId: layoutId,
+    });
+
+    expect(prodLayout).to.be.ok;
+    expect(prodLayout._parentId).to.eql(devLayout._id);
+    expect(prodLayout._environmentId).to.eql(prodEnv._id);
+    expect(prodLayout._organizationId).to.eql(session.organization._id);
+    expect(prodLayout._creatorId).to.eql(session.user._id);
+    expect(prodLayout.name).to.eql(layoutName);
+    expect(prodLayout.content).to.eql(content);
+    // TODO: Awful but it comes from the repository directly.
+    const { _id: _, ...prodVariables } = prodLayout.variables?.[0] as any;
+    expect(prodVariables).to.deep.include(variables[0]);
```

Comment:
Might be annoying but worth maybe to check more detailed comparison of the objects? Up to your choice.